include/blob/blob_client.h

```diff
@@ -400,7 +412,8 @@ namespace azure { namespace storage_lite {
         /// <param name="blob">The blob name.</param>
         /// <param name="is">The source stream.</param>
         /// <param name="metadata">A <see cref="std::vector"> that respresents metadatas.</param>
-        void upload_block_blob_from_stream(const std::string &container, const std::string blob, std::istream &is, const std::vector<std::pair<std::string, std::string>> &metadata = std::vector<std::pair<std::string, std::string>>());
+        /// <param name="streamlen">Length of the stream. Used only when the stream does not support tellg/seekg</param>
+        void upload_block_blob_from_stream(const std::string &container, const std::string blob, std::istream &is, const std::vector<std::pair<std::string, std::string>> &metadata = std::vector<std::pair<std::string, std::string>>(), size_t streamlen=0);
```

Comment:
format of "streamlen=0", please add extra space just like other parameters.

Reply:
Use 0 to indicate user set the len or not is tricky. What if user used 0 but actually he/she wants to create an empty blob?

Reply:
The other way is to compare it to the max value. Set the default value to ULLONG_MAX.